src/transform-sdk/rust/sr/src/lib.rs

```diff
@@ -145,6 +145,97 @@ impl<Key: HashT + Eq, Value: Clone> LruCache<Key, Value> {
         self.underlying.borrow_mut().put(k, v)
     }
 }
+
+#[derive(Debug)]
+struct CachingSchemaRegistryClient {
+    underlying: AbiSchemaRegistryClient,
+    schema_by_id_cache: LruCache<SchemaId, Schema>,
+    schema_by_sv_cache: LruCache<SubjectVersion, SubjectSchema>,
+}
+
+/// SubjectVersion is an aggregate type for the registered subject and version of a schema
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+struct SubjectVersion {
+    subject: String,
+    version: SchemaVersion,
+}
+
+impl SubjectVersion {
+    pub fn new(subject: impl Into<String>, version: SchemaVersion) -> Self {
+        Self {
+            subject: subject.into(),
+            version,
+        }
+    }
+}
+
+#[allow(dead_code)]
+impl CachingSchemaRegistryClient {
+    pub fn new(max_entries: Option<usize>) -> CachingSchemaRegistryClient {
+        // No need for an abi::check() here; that's done in AbiSchemaRegistryClient::new()
+        Self {
+            underlying: AbiSchemaRegistryClient::new(),
+            schema_by_id_cache: LruCache::<SchemaId, Schema>::new(max_entries),
+            schema_by_sv_cache: LruCache::<SubjectVersion, SubjectSchema>::new(max_entries),
+        }
+    }
+}
+
+impl SchemaRegistryClientImpl for CachingSchemaRegistryClient {
+    fn lookup_schema_by_id(&self, id: SchemaId) -> Result<Schema> {
+        if let Some(schema) = self.schema_by_id_cache.get(&id) {
+            Ok(schema.clone())
```

Comment:
ah, i guess we're cloning these out anyway. now cache::get will return a value type so we can just relocate into the Result.